packages/shared/src/guards.test.ts

```diff
@@ -133,27 +133,100 @@ describe('assertDatasetValue', () => {
         dataset('foo', intType(true, 64), [2]),
       ),
     ).not.toThrow();
+
+    expect(() =>
+      assertDatasetValue(
+        Float32Array.from([0, 1]), // big ints can be returned as any kind of numbers
+        dataset('foo', intType(true, 64), [2]),
+      ),
+    ).not.toThrow();
   });
 
-  describe('assertDatasetValue', () => {
-    it("should throw when value doesn't satisfy dataset type and shape", () => {
-      expect(() =>
-        assertDatasetValue(
-          true,
-          dataset('foo', enumType(intType(), { FOO: 0 }), []),
-        ),
-      ).toThrow('Expected number');
-
-      expect(() =>
-        assertDatasetValue(['foo', 'bar'], dataset('foo', intType(), [2])),
-      ).toThrow('Expected number');
-
-      expect(() =>
-        assertDatasetValue(
-          BigInt64Array.from([0n, 1n]),
-          dataset('foo', intType(), [2]),
-        ),
-      ).toThrow('Expected number');
-    });
+  it("should throw when value doesn't satisfy dataset type and shape", () => {
+    expect(() =>
+      assertDatasetValue(
+        true,
+        dataset('foo', enumType(intType(), { FOO: 0 }), []),
+      ),
+    ).toThrow('Expected number');
+
+    expect(() =>
+      assertDatasetValue(['foo', 'bar'], dataset('foo', intType(), [2])),
+    ).toThrow('Expected number');
+
+    expect(() =>
+      assertDatasetValue(
+        BigInt64Array.from([0n, 1n]),
+        dataset('foo', intType(), [2]),
+      ),
+    ).toThrow('Expected number');
+  });
+
+  it('should not throw when value shape satisfies selection', () => {
+    expect(() =>
+      assertDatasetValue(
```

Comment:
`assertDatasetValue` now takes a `selection` string and asserts that the value is a scalar when a scalar selection is passed.